test/e2e/storage/persistent_volumes.go

```diff
@@ -308,45 +310,109 @@ var _ = utils.SIGDescribe("PersistentVolumes", func() {
 				framework.SkipUnlessProviderIs("openstack", "gce", "gke", "vsphere", "azure")
 
 				numVols := 4
-				pvcs := []*v1.PersistentVolumeClaim{}
+				ssTester := framework.NewStatefulSetTester(c)
 
-				By("Creating PVCs")
+				By("Creating a StatefulSet pod to initialize data")
+				writeCmd := "true"
 				for i := 0; i < numVols; i++ {
-					pvc = framework.MakePersistentVolumeClaim(framework.PersistentVolumeClaimConfig{}, ns)
-					pvc, err = framework.CreatePVC(c, ns, pvc)
-					Expect(err).NotTo(HaveOccurred())
-					pvcs = append(pvcs, pvc)
+					writeCmd += fmt.Sprintf("&& touch %v", getVolumeFile(i))
 				}
-
-				By("Waiting for PVCs to be bound")
-				for _, pvc := range pvcs {
-					framework.Logf("Created PVC %q", pvc.Name)
-					framework.ExpectNoError(framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, c, ns, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout))
+				writeCmd += "&& sleep 10000"
+
+				probe := &v1.Probe{
+					Handler: v1.Handler{
+						Exec: &v1.ExecAction{
+							// Check that the last file got created
+							Command: []string{"test", "-f", getVolumeFile(numVols - 1)},
+						},
+					},
+					InitialDelaySeconds: 1,
+					PeriodSeconds:       1,
 				}
 
-				By("Creating a pod and initializing data")
-				writeCmd := "true"
-				for i, pvc := range pvcs {
-					// mountPath is /mnt/volume<i+1>
-					writeCmd += fmt.Sprintf("&& touch /mnt/volume%v/%v", i+1, pvc.Name)
+				mounts := []v1.VolumeMount{}
+				claims := []v1.PersistentVolumeClaim{}
+				for i := 0; i < numVols; i++ {
+					pvc := framework.MakePersistentVolumeClaim(framework.PersistentVolumeClaimConfig{}, ns)
+					pvc.GenerateName = ""
```

Comment:
do we need to setting this explicitly to ""

Reply:
nope i've removed